moveit_ros/hybrid_planning/global_planner/global_planner_plugins/src/moveit_planning_pipeline.cpp

```diff
@@ -97,45 +100,102 @@ moveit_msgs::msg::MotionPlanResponse MoveItPlanningPipeline::plan(
     return response;
   }
 
-  // Process goal
-  if ((global_goal_handle->get_goal())->motion_sequence.items.size() > 1)
+  // If global_traj_pass_through_ and multiple waypoints were requested, assume we have a fully planned trajectory
+  // already. Just forward it to the local planner. This works only for a JointConstraint type.
+  size_t num_waypoints = global_goal_handle->get_goal()->motion_sequence.items.size();
+  if (global_traj_pass_through_ && num_waypoints > 1)
   {
-    RCLCPP_WARN(LOGGER, "Global planner received motion sequence request with more than one item but the "
-                        "'moveit_planning_pipeline' plugin only accepts one item. Just using the first item as global "
-                        "planning goal!");
+    auto motion_plan_req = global_goal_handle->get_goal()->motion_sequence.items[0].req;
+    // Start state is the current state of the robot, by default
+    // Assume these parameters don't change for any waypoint, aka sequence item
+    response.group_name = motion_plan_req.group_name;
+    response.planning_time = motion_plan_req.allowed_planning_time;
+    response.error_code.val = moveit_msgs::msg::MoveItErrorCodes::SUCCESS;
+    moveit::core::RobotStatePtr current_state;
+    if (!moveit_cpp_->getCurrentState(current_state, 0.5))
+    {
+      RCLCPP_ERROR(LOGGER, "Could not retrieve the current robot state.");
+      std::exit(EXIT_FAILURE);
+    }
+    moveit_msgs::msg::RobotState current_state_msg;
+    moveit::core::robotStateToRobotStateMsg(*current_state, current_state_msg);
+    response.trajectory_start = current_state_msg;
+
+    moveit_msgs::msg::Constraints goal_constraints =
+        global_goal_handle->get_goal()->motion_sequence.items.at(0).req.goal_constraints.at(0);
+    if (goal_constraints.joint_constraints.empty())
+    {
+      RCLCPP_ERROR(LOGGER, "Constraints of the motion plan request were empty or not of JointConstraint type.");
+      std::exit(EXIT_FAILURE);
+    }
+
+    // Get joint names
+    for (const auto& constraint : goal_constraints.joint_constraints)
+    {
+      response.trajectory.joint_trajectory.joint_names.push_back(constraint.joint_name);
+    }
+
+    // Get joint positions
```

Comment:
This is not really how the constraints are supposed to be used and would break some use cases of the normal planning pipeline. If you really want to forward a predefined trajectory, better adapt the planner logic and publish it to the local planner directly, no need to implement pass-through behavior